Mathlib/NumberTheory/GaussSum.lean

```diff
@@ -158,11 +158,10 @@ theorem gaussSum_frob (χ : MulChar R R') (ψ : AddChar R R') :
 /-- For a quadratic character `χ` and when the characteristic `p` of the target ring
 is a unit in the source ring, the `p`th power of the Gauss sum of`χ` and `ψ` is
 `χ p` times the original Gauss sum. -/
--- Porting note: Added `nonrec` to avoid error `failed to prove termination`
-nonrec theorem MulChar.IsQuadratic.gaussSum_frob (hp : IsUnit (p : R)) {χ : MulChar R R'}
+theorem MulChar.IsQuadratic.gaussSum_frob (hp : IsUnit (p : R)) {χ : MulChar R R'}
```

Comment:
This is what `nonrec` is for, so I'm not sure we should avoid it.

Reply:
I was thinking of [this Zulip message](https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/.E2.80.9Cdeprecated.E2.80.9D.20error.20message/near/438479016) here, but I have no strong opinion.